app/controllers/ride_mechanics_controller.rb

```diff
@@ -0,0 +1,7 @@
+class RideMechanicsController < ApplicationController
+  def create
+    RideMechanic.create!(mechanic: Mechanic.find(params[:mechanic_id]), ride: Ride.find(params[:ride_id]))
+
+    redirect_to(mechanic_path(Mechanic.find(params[:mechanic_id])))
+  end
+end
```

Comment:
Careful with adding `!` (bangs) into your code. They should only be used if you are using exception handlers. Otherwise, if for whatever reason they fail, it will render a really ugly error to our end user.

On line 5, you do not need to find the Mechanic again. As long as you pass the path helpers an id, they will be able to generate the correct path. So, this could look something like: ` redirect_to(mechanic_path(params[:mechanic_id]))`